flytekit/core/context_manager.py

```diff
@@ -741,6 +741,17 @@ def get_deck(self) -> typing.Union[str, "IPython.core.display.HTML"]:  # type:ig
 
         return _get_deck(typing.cast(ExecutionState, self.execution_state).user_space_params)
 
+    def add_deck(self, deck: Deck) -> None:
```

Comment:
Although this is interesting to have, can this method be removed from this PR? It's orthogonal to adding a Python Dependency Deck and decks are automatically added to the context during `Deck.__init__`:

https://github.com/flyteorg/flytekit/blob/3642ec6771985f504c2b6a9dfd2598a84219ac58/flytekit/deck/deck.py#L58

As for this PR, I'll want to include the deck automatically when `enable_deck=True` and add `PythonDependencyDeck` here:

https://github.com/flyteorg/flytekit/blob/3642ec6771985f504c2b6a9dfd2598a84219ac58/flytekit/core/python_function_task.py#L351

Reply:
Good advice, thanks

Reply:
updated